backend/amd64/simd_selection.ml

```diff
@@ -400,21 +408,22 @@ let register_behavior_sse = function
 
 let register_behavior_sse2 = function
   | Add_i8 | Add_i16 | Add_i32 | Add_i64 | Add_f64 | Add_saturating_i8
-  | Add_saturating_i16 | Add_saturating_unsigned_i8
-  | Add_saturating_unsigned_i16 | Sub_i8 | Sub_i16 | Sub_i32 | Sub_i64 | Sub_f64
-  | Sub_saturating_i8 | Sub_saturating_i16 | Sub_saturating_unsigned_i8
-  | Sub_saturating_unsigned_i16 | Max_unsigned_i8 | Max_i16 | Max_f64
-  | Min_unsigned_i8 | Min_i16 | Min_f64 | Mul_f64 | Div_f64 | And_bits
-  | Andnot_bits | Or_bits | Xor_bits | Cmpeq_i8 | Cmpeq_i16 | Cmpeq_i32
-  | Cmpgt_i8 | Cmpgt_i16 | Cmpgt_i32 | Cmp_f64 _ | SLL_i16 | SLL_i32 | SLL_i64
-  | SRL_i16 | SRL_i32 | SRL_i64 | SRA_i16 | SRA_i32 | Avg_unsigned_i8
+  | Min_scalar_f64 | Max_scalar_f64 | Add_saturating_i16
+  | Add_saturating_unsigned_i8 | Add_saturating_unsigned_i16 | Sub_i8 | Sub_i16
+  | Sub_i32 | Sub_i64 | Sub_f64 | Sub_saturating_i8 | Sub_saturating_i16
+  | Sub_saturating_unsigned_i8 | Sub_saturating_unsigned_i16 | Max_unsigned_i8
+  | Max_i16 | Max_f64 | Min_unsigned_i8 | Min_i16 | Min_f64 | Mul_f64 | Div_f64
+  | And_bits | Andnot_bits | Or_bits | Xor_bits | Cmpeq_i8 | Cmpeq_i16
+  | Cmpeq_i32 | Cmpgt_i8 | Cmpgt_i16 | Cmpgt_i32 | Cmp_f64 _ | SLL_i16 | SLL_i32
+  | SLL_i64 | SRL_i16 | SRL_i32 | SRL_i64 | SRA_i16 | SRA_i32 | Avg_unsigned_i8
   | Avg_unsigned_i16 | SAD_unsigned_i8 | Shuffle_64 _ | Interleave_high_8
   | Interleave_high_16 | Interleave_high_64 | Interleave_low_8
   | Interleave_low_16 | Interleave_low_64 | I16_to_i8 | I32_to_i16
   | I16_to_unsigned_i8 | I32_to_unsigned_i16 ->
     R_RM_to_fst
   | Shuffle_high_16 _ | Shuffle_low_16 _ | I32_to_f64 | I32_to_f32 | F64_to_i32
-  | F64_to_f32 | F32_to_i32 | F32_to_f64 ->
+  | Cast_scalar_f64_i64 | F64_to_f32 | F32_to_i32 | F32_to_f64 | Sqrt_scalar_f64
+    ->
```

Comment:
Previously, `sqrt` was constraint to registers as arg and res. It can take arg in memory so your change is fine, but please make sure to have a specific test for this.

Reply:
What's the best way to test this? The operand won't get spilled unless there is particular register pressure, but even in that case it seems to prefer loading the operand to `xmm0` and then doing `sqrtsd xmm0,xmm0`

Reply:
not sure, @xclerc  do you have a suggestion? 

if it's hard to test, perhaps we should keep is as `R_to_R` in `register_behavior` in this PR, to make progress, and make this change in a separate PR (would be a good idea anyway)?

Reply:
Switched to `R_to_R`